x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -43,18 +45,17 @@ h| Information     | ID Token   | Access Token
 In addition, {es} also validates `nbf` and `auth_time` claims for ID Tokens if these claims are present.
 But these claims are ignored for access tokens.
```

Comment:
> Is the validation skipped because we could find claim formats other than integers? How complex would be to parse a datetime field and convert it?

The format is definitely one of the main reasons. But more fundamentally is that we don't have a popular standard to work with. We can try to parse different formats and validate as much as possible. But doing so does not conform to any specs (rfc9068 and oidc expect the time claim to be integers). It also feels weird to do so just for `auth_time` of access_token type, but not for `iat` and `exp` of any token types. In the case when the value turns out to be not a timestamp after all, our best bet is probably just skip the claim instead of authentication failure. If so, the benefit of validation is further reduced and harder to explain, e.g. the value of `2023-01-01 00:00:00Z` may not pass the validation and fails authentication because the timestamp is too old, but if it has a typo and hence cannot be parsed as a timestamp, e.g. `2023-01-01_00:00:00Z`, it is skipped and authentication succeeds.

Between `auth_time` and `nbf`, validating`auth_time` is not very useful because it can only be before `iat`. Since we already mandate `iat` must be before "now", validating the same thing for `auth_time` does not really achieve anything. I can see some value in validating `nbf` because it can be after `iat` thus further restricts when the JWT can be used. But overall I still prefer not bother with it for access_token type for simplicity. Its use case is not very prominent and feels more suitable for a future enhancement request when we have concrete asks.